service/controller/azureconfig/handler/capzcrs/create.go

```diff
@@ -104,10 +104,10 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	return nil
 }
 
-func (r *Resource) mapAzureConfigToCluster(ctx context.Context, cr providerv1alpha1.AzureConfig, infraRef *corev1.ObjectReference) (runtime.Object, error) {
-	cluster := &capiv1alpha3.Cluster{
+func (r *Resource) mapAzureConfigToCluster(ctx context.Context, cr providerv1alpha1.AzureConfig, infraRef *corev1.ObjectReference) (client.Object, error) {
+	cluster := &capi.Cluster{
 		TypeMeta: metav1.TypeMeta{
```

Comment:
Every time I see this `TypeMeta` I wonder: is this really necessary? Could we drop it? It's used *extensively* on `kubectl-gs` and it makes it harder to migrate to future api versions.